diff --git a/arch/arm64/boot/dts/rockchip/rk3399-hm3399.dts b/arch/arm64/boot/dts/rockchip/rk3399-hm3399.dts
index afdfea0cc..8528306b8 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-hm3399.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3399-hm3399.dts
@@ -150,7 +150,7 @@ sdio_pwrseq: sdio-pwrseq {
 		 * - SDIO_RESET_L_WL_REG_ON
 		 * - PDN (power down when low)
 		 */
-		reset-gpios = <&gpio2 27 GPIO_ACTIVE_HIGH>; /* GPIO2_D3 */
+		reset-gpios = <&gpio2 RK_PD3 GPIO_ACTIVE_HIGH>; /* GPIO2_D3 */
 	};
 	
 	wireless-wlan {
@@ -158,7 +158,7 @@ wireless-wlan {
 		rockchip,grf = <&grf>;
 		wifi_chip_type = "ap6335";
 		sdio_vref = <1800>;
-		WIFI,host_wake_irq = <&gpio0 3 GPIO_ACTIVE_HIGH>; /* GPIO0_a3 */
+		WIFI,host_wake_irq = <&gpio0 RK_PA3 GPIO_ACTIVE_HIGH>; /* GPIO0_a3 */
 		status = "okay";
 	};
 
@@ -248,7 +248,7 @@ fusb0: fusb30x@22 {
 		compatible = "fairchild,fusb302";
 		reg = <0x22>;
 		pinctrl-names = "default";
-		pinctrl-0 = <&fusb0_int>;
+		pinctrl-0 = <&fusb0_int &fusb0_vbus>;
 		int-n-gpios = <&gpio1 RK_PA2 GPIO_ACTIVE_HIGH>;//GPIO1_A2
 		vbus-5v-gpios = <&gpio1 RK_PC4 GPIO_ACTIVE_HIGH>;//GPIO1_C4
 		status = "okay";
@@ -273,7 +273,7 @@ &pcie0 {
 &pinctrl {
 	sdio-pwrseq {
 		wifi_enable_h: wifi-enable-h {
-			rockchip,pins = <0 RK_PB2 RK_FUNC_GPIO &pcfg_pull_none>;
+			rockchip,pins = <2 RK_PD3 RK_FUNC_GPIO &pcfg_pull_none>;
 		};
 	};
 
diff --git a/arch/arm64/boot/dts/rockchip/rk3399-hm3399.dtsi b/arch/arm64/boot/dts/rockchip/rk3399-hm3399.dtsi
index 001ed359e..71eaa3cbc 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-hm3399.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3399-hm3399.dtsi
@@ -90,22 +90,7 @@ vcc5v0_host: vcc5v0-host-regulator {
 		regulator-always-on;
 		vin-supply = <&vcc_sys>;
 	};
-
-	vcc5v0_typec0: vcc5v0-typec0-regulator {
-		compatible = "regulator-fixed";
-		enable-active-high;
-		gpio = <&gpio1 RK_PC4 GPIO_ACTIVE_HIGH>;//HUB_RST   GPIO1_C4
-		pinctrl-names = "default";
-		pinctrl-0 = <&vcc5v0_typec0_en>;
-		regulator-name = "vcc5v0_typec0";
-		vin-supply = <&vcc_sys>;
-		
-		regulator-state-mem {
-			regulator-off-in-suspend;
-		};
-	};
 	
-
 	vcc_sys: vcc-sys {
 		compatible = "regulator-fixed";
 		regulator-name = "vcc_sys";
@@ -135,8 +120,8 @@ vdd_log: vdd-log {
 		compatible = "5v_en";
 		pinctrl-names = "default";
 		pinctrl-0 = <&pwr4g_pin &rst4g_pin>;
-		hub_rst = <&gpio4 RK_PD5 GPIO_ACTIVE_LOW>;
-		pwr_led_en = <&gpio1 RK_PA0 GPIO_ACTIVE_LOW>;
+		//hub_rst = <&gpio4 RK_PD5 GPIO_ACTIVE_LOW>;
+		//pwr_led_en = <&gpio1 RK_PA0 GPIO_ACTIVE_LOW>;
 		pwr_en_4g = <&gpio1 RK_PD0 GPIO_ACTIVE_HIGH>;//gpio1_d0
 		rst_4g = <&gpio4 RK_PD4 GPIO_ACTIVE_HIGH>;//gpio4_d4
 		status = "okay";
@@ -531,18 +516,18 @@ sdmmc0_pwr_h: sdmmc0-pwr-h {
 	usb2 {
 		vcc5v0_host_en: vcc5v0-host-en {
 			rockchip,pins =
-				<4 29 RK_FUNC_GPIO &pcfg_pull_up>;
-		};
-		vcc5v0_typec0_en: vcc5v0-typec0-en {
-			rockchip,pins =
-				<1 20 RK_FUNC_GPIO &pcfg_pull_up>;
+				<4 RK_PD5 RK_FUNC_GPIO &pcfg_pull_up>;
 		};
 	};
 	
 	fusb30x {
 		fusb0_int: fusb0-int {
-			rockchip,pins = <1 2 RK_FUNC_GPIO &pcfg_pull_up>;
-		};			
+			rockchip,pins = <1 RK_PA2 RK_FUNC_GPIO &pcfg_pull_up>;
+		};	
+		
+		fusb0_vbus: fusb0-vbus {
+			rockchip,pins = <1 RK_PC4 RK_FUNC_GPIO &pcfg_pull_up>;
+		};		
 
 	};
 	
@@ -719,7 +704,7 @@ u2phy0_otg: otg-port {
 	};
 
 	u2phy0_host: host-port {
-		phy-supply = <&vcc5v0_typec0>;
+		phy-supply = <&vcc5v0_host>;
 		status = "okay";
 	};
 };
