diff --git a/sound/soc/soc-dapm.c b/sound/soc/soc-dapm.c
index c0aa64ff8..deada1d5f 100644
--- a/sound/soc/soc-dapm.c
+++ b/sound/soc/soc-dapm.c
@@ -4741,7 +4741,7 @@ int snd_soc_dapm_ignore_suspend(struct snd_soc_dapm_context *dapm,
 	struct snd_soc_dapm_widget *w = dapm_find_widget(dapm, pin, false);
 
 	if (!w) {
-		dev_err(dapm->dev, "ASoC: unknown pin %s\n", pin);
+		/*dev_err(dapm->dev, "ASoC: unknown pin %s\n", pin);*/
 		return -EINVAL;
 	}
 
